Moved outstanding exp-func-imp PR into this PR with latest reorg

larrybabb opened the pull request:
This is the draft for the latest ExpFuncImpact profiles applied to the latest reorg of folders that came out of the ASHG unconference work.

Ready for review, refine, approve and merge after @ahwagner, @mbrush and @larrybabb meet on 11/8/2024.

larrybabb committed:
Moved outstanding exp-func-imp PR into this PR with latest reorg of folders

larrybabb committed:
updates from nov.07.2024 gks-tech sync with MB/AW/LB

larrybabb commented:
@mbrush and @ahwagner please review this and let's push it if it's clean enough based on all conversations/decisions to date. We can continue with another PR for the next tranche of changes.

Notes from gks-tech sync mtg 11/7

1. change 'type' to 'groupType' on TherapyGroup
2. remove 'type on TraitSet. Use extensions
3. keep 'base'
4. rename all files with full names (sans variant)
  i.e cohort-allele-frequency-study-result-profile-source.yaml
5. rename shared-concept-source.yaml to domain-entities-source.yaml
6. delete assay-var-efect-source.yaml (double check nothing was lost)
    (convert tests)
7. pathogenicity-evidenceline-profile-source.yaml (with proposition in it)
8. make EvidenceLine heritable

larrybabb committed:
removed notes file

larrybabb commented:
I added a mock-up of a VarPath statement in the `acmg-2015` schema subfolder that builds for the next area of discussion.

larrybabb commented on schema/va-spec/base/pathogenicity-evidenceline-profile-source.yaml:
```diff
@@ -0,0 +1,100 @@
+
+$defs:
+  # Pathogenicity Evidence Line
+  PathogenicityEvidenceLine:    
```
@mbrush I'm curious if you intentionally left `Variant` out of the name of this class. We have `VariantPathogenicityStatement` and `VariantPathogenicityProposition` but this class is just `PathogenicityEvidenceLine` instead of `VariantPathogenicityEvidenceLine`? It seems inconsistent.

mbrush replied:
Agreed.  That was just an oversight.  We should call it VariantPathogenicityEvidenceLine.

larrybabb commented on schema/va-spec/base/pathogenicity-evidenceline-profile-source.yaml:
```diff
@@ -0,0 +1,100 @@
+          pathogenicity of the assessed variant, or remain neutral)
+      strengthOfEvidenceProvided:           # TO DO: Ensure that ten enum below covers terms needed for initial adopters. Merge https://github.com/ga4gh/gks-common/pull/78 which updates the core IM to allow this property to take a string. Think about if we need the PS/BS codes here - as this bakes in direction, and is duplicative as the 'specifiedBy' attribute reports the ACMG criterion applied. 
+        extends: strengthOfEvidenceProvided
+        type: string
+        enum:
+          - PS3_Strong
+          - PS3_Moderate
+          - PS3_Supporting
+          - BS3_Strong
+          - BS3_Moderate
+          - BS3_Supporting
```
This needs to be changed for several reasons:
1. If it truly meant to hold the specific MappableConcepts for each of the ACMG-2015 codes then it should be moved into the specialized `VariantPathogenicityEvidenceLine` in the `ACMG-2015` folder/namespace and be made more comprehensive. 
2. The `EvidenceLine.strengthOfEvidenceProvided` attribute in `gks-core` is of type `MappableConcept` at this time. So we either need to change that to a string or change this to a `MappableConcept` or do something completely different that will align the datatypes in both?  (possibly creating a specialized class or start making a plan for `ValueSet`s)

ahwagner replied:
I agree with these points. I might add that we may be better off using a shared, generic, concrete EvidenceLine class that contains all of the attributes seen in the (Variant)PathogenicityEvidenceLine, which is constrained within the ACMG-2015 namespace. Unless we think there are entire properties that are specific to all (Variant)PathogenicityEvidenceLines but not other evidence lines (e.g. AMP/ASCO/CAP evidence lines).

larrybabb replied:
@mbrush and @larrybabb will work through the plan to add an `evidenceOutcome` property on `EvidenceLine` which will summarize the met or not met criteria derived from the methods.

larrybabb replied:
moved to new PR

larrybabb committed:
Completed converting pre-existing AVE test cases with new ExpVarFuncImpact profiles. Some questions remain (see comments in test case fixtures)

larrybabb commented on schema/va-spec/acmg-2015/pathogenicity-statement-profile-source.yaml:
```diff
@@ -0,0 +1,32 @@
+    - $ref: "/ga4gh/schema/va-spec/1.x/base/VariantPathogenicityStatement"
+    - properties:
+        classification:
+          # this is a MappableConcept but we would ideally like to specify a value set with meta characteristics to validate with some consistency
```
We should discuss this general concern when patterns like this occur. And they will occur fairly often during profiling.

this is a MappableConcept but we would ideally like to specify a value set with meta characteristics to validate with some consistency

ahwagner replied:
I'm not following; if we uncomment the code below, does this not do what we want? What are the meta characteristics you are describing here?

larrybabb replied:
moved to new PR

larrybabb committed:
reverted premature`trial use` maturity values to `draft`

larrybabb commented on schema/va-spec/base/experimental-functional-impact-study-statement-profile-source.yaml:
```diff
@@ -0,0 +1,101 @@
+        extends: object
+        oneOf:
+          - $ref: "/ga4gh/schema/gks-core/1.x/json/iriReference"
+          - $ref: "/ga4gh/schema/gks-core/1.x/MappableConcept"      # See https://github.com/ga4gh/gks-common/pull/77
```
@mbrush can we archive/remove the comment on the end of this line or do we need to still visit the proposal of creating a `SequenceFeature` class?

ahwagner replied:
I agree, let's drop the comment and notion of creating a SequenceFeature class for now. `MappableConcept` should cover us for our immediate use case.

larrybabb replied:
sticking with MappableConcept for now. Will revisit specialized domain entities as we evolve.

larrybabb commented on schema/va-spec/base/experimental-functional-impact-study-statement-profile-source.yaml:
```diff
@@ -0,0 +1,101 @@
+          functional impact(s) measured by the assay(s) employed. It is recommended that terms
+          are drawn from established terminologies where possible.
+          
+        # TO DO: I feel like it would be good to try and standardize the values used here by defining an enum based on early adopter needs, and growing from there. 
```
TODO see comment here.

ahwagner replied:
Agree; before adding this property we should identify what is envisioned to fit here.

larrybabb replied:
address during 11-13 sync up. we decided it may be premature to lock down the set of terms to be used by both mave and civic for the impactTypeQualifier. If we choose to add this it would need to live in the community profile. We have moved this and commented out how this could be done for future consideration.

larrybabb commented on schema/va-spec/base/experimental-functional-impact-study-statement-profile-source.yaml:
```diff
@@ -0,0 +1,101 @@
+        extends: subject
+        oneOf:
+          - $ref: "/ga4gh/schema/vrs/2.x/json/MolecularVariation"
+          - $ref: "/ga4gh/schema/cat-vrs/1.x/json/CategoricalVariant"
```
@ahwagner this is sort of interesting and odd and hard to explain IMO. CategoricalVariant is vast in terms of what can be represented but MolecularVariation is trying to limit it to just contextual variants on a molecule. Should we be considering whether or not we provide some way to group `systemic` vs `molecular` catVars?  If it is important at the context VRS level it seems like it would also have the same or similar importance in cat-vrs...right?

ahwagner replied:
It's an interesting thought, but not a problem we need to solve immediately. Easy to add constraints later.

However, I wonder if we would be doing ourselves a favor by leaving this a little broader right now, allow for any VRS variation, not just molecular.

larrybabb commented on schema/va-spec/base/pathogenicity-evidenceline-profile-source.yaml:
```diff
@@ -0,0 +1,129 @@
+      - predicate
+      - objectCondition
+
+     # TO DO: decide if we want to include the qualifiers defined on the VarPathStatement class in the VarPathProposition
```
@mbrush I went for it and added in the `qualifiers` from VarPathStatement so we could see it first-hand. It starts making things pretty clear that we "probably" will move to using 'targetProposition' on `Statement` and forego the flattening of these fields directly into Statement. As it stands we will need to always make sure this schema stays in lock step with it's sibling Statement SPOQ attributes specs. (TRICKY!) ... @ahwagner should we?

ahwagner replied:
Holy crap. If I am understanding you right, @larrybabb, you just sold me on this. I think that you are suggesting that we move ALL of the SPOQ attributes to propositions, making these propositions reusable across statement types and evidence lines, with the consequence that BOTH evidence lines and statement classes become concrete entities that don't need subclassing (because all of the domain-specific properties live in the typed proposition).

Am I understanding that right!? If so, this is a solid design improvement and I am in favor.

larrybabb replied:
This was agreed upon by all of us at the 11-12 tech sync up. And it has been applied to 1.x and in the new PR.

larrybabb commented on tests/fixtures/AVE-clinical-classification.yaml:
```diff
@@ -1,36 +1,60 @@
+    #     but I would imagine it may make debugging validation failures much easier
+    #     if the categoryType or recipeType was stated explicitly in the object.
+    type: CategoricalVariant
+    # subtype: ProteinSequenceConsequence  or categoryType: ProteinSequenceConsequence
```
@ahwagner  A worthy discussion here IMO. +1 from me for adding a `categoryType` or `subtype` to CategoricalVariant.

ahwagner replied:
I think this boils down to a domain issue. I get that if you are only considering a handful of common patterns (e.g. three categories of Canonical Allele, Canonical CNV, and "other stuff"), ideally you would want these as named classes, similar to how we do things in VRS.

I just want to caution against the idea that we rely on typing for top-level concepts Cat-VRS. Typed constraints should get you everything you need and scale far better in this problem space. I know I have gone over the [Somatic Variation Landscape](https://docs.google.com/presentation/d/1KvHcyaLCMJ6HDlKsTGSKYoICBSt0Utdncsq1FQRGcIM/edit#slide=id.p) before, but these are just a handful of what are a combinatorial number of potential types of categorical variants based on the various ways different constraints are used. Should I build an implementation that looks for a subtype based on a combinatorial list of types that conveys an object has a `gene constraint + functional constraint`, or a `gene constraint + location constraint + functional constraint`, or a `gene constraint + location constraint + molecular constraint`, or a `location constraint + molecular constraint`,  or the various combinations of input configurations of those constraints (e.g. `location constraint with "liftover_to" relation`), _ad nauseum_? I hope not. When consuming these data for use in a system that covers the broad heterogeneity of concept types seen in the VICC knowledgebases, I would instead rely on the types of functional constraints for processing. Ultimately, I would think you would want the same thing, once you start digging into what that "other stuff" in ClinVar is.

It seems to me like the current setup generalizes cross-domain concerns and saves us the difficulty of explicitly naming the many potential combinations of constraints and constraint properties. Building our systems in the absence of such named category types seems like the right mindset for generalized software in this domain.

Do you disagree? If so, I recommend we move this conversation to a discussion thread in Cat-VRS and hash it out there.

larrybabb replied:
moved this to https://github.com/ga4gh/cat-vrs/discussions/81

larrybabb commented on tests/fixtures/AVE-clinical-classification.yaml:
```diff
@@ -1,36 +1,60 @@
+  geneContextQualifier:
+    conceptType: Gene
+    label: PTEN
+# LB: the following objectAssay was in the original test example. I assume it should be dropped, please verify.
```
TODO @mbrush can you verify that Alan's old test case was using objectAssay in this test case and you guys decided it does not belong at this level. If it does need to get moved to a different attribute please let me know so I can either fix it or remove it.

larrybabb replied:
nvm ☝️

larrybabb commented on tests/fixtures/AVE-clinical-classification.yaml:
```diff
@@ -1,36 +1,60 @@
+
+# LB: strengthOfEvidenceProvided is illogically specified in PathogenicityEvidenceLine 
+#     to conflict with the underlying datatype of MappableConcept 
+# strengthOfEvidenceProvided: 
```
TODO this one is breaking the test case. We won't be able to comment out the real stuff so we must resolve this.

ahwagner replied:
Will take a look at this myself when running tests to understand what the issue is.

ahwagner replied:
resolved in e98bda5.

larrybabb reviewed (commented):
I'm good on this.  I'd like to address all this on Fri Nov 8 and then merge this. We should determine what's beyond the scope covered by this PR and target that for the next PR.

mbrush committed:
Rename pathogenicity-evidenceline-profile-source.yaml to pathogenicity-evidence-line-profile-source.yaml

adding dash to evidence-line file name to be consistent with sue of dashes between class names in other profile file names (e.g. study-result)

mbrush committed:
Update experimental-functional-impact-study-statement-profile-source.yaml

Updated comments for discussion/review among developers, based on most recent conversations.

mbrush committed:
Update pathogenicity-evidence-line-profile-source.yaml

- Updated comments/questions about this profile based on outcomes of recent discussions.
- made a couple minor improvements to attribute descriptions.

ahwagner committed:
rebuild rst

ahwagner committed:
resolves conflicting definition of evidence line property

ahwagner committed:
update build with newer MSP

ahwagner committed:
update fixtures to new mappable concept

ahwagner committed:
add support for namespaced tests

ahwagner committed:
Merge branch '1.x' into func-impact-for-review

ahwagner committed:
fix documentation refs for statement profiles

ahwagner committed:
reorg documentation for base and community profiles

ahwagner committed:
draft solution to ga4gh/gks-metaschema#35

larrybabb committed:
Replace Stmt SPOQ attribs with Proposition.

larrybabb committed:
add back lost proposal from AHW on community docs

larrybabb committed:
draft changes to acmg-2015 func-imp-evid-line

larrybabb committed:
clean schema folders
